packages/eslint-plugin/src/rules/consistent-type-exports.ts

```diff
@@ -109,6 +109,7 @@ export default createRule<Options, MessageIds>({
       ExportNamedDeclaration(node: TSESTree.ExportNamedDeclaration): void {
         // Coerce the source into a string for use as a lookup entry.
         const source = getSourceFromExport(node) ?? 'undefined';
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
This is the old issue we've got filed right where TS is giving the wrong type for logical assignment operators?

If yes - Oof we should prioritise this #6762

Reply:
Yeah :/